ansible/roles/vm_set/tasks/add_topo.yml

```diff
@@ -52,6 +52,17 @@
           - net_admin
         privileged: yes
       become: yes
+
+    - name: Bind ptf_ip to keysight_api_server
```

Comment:
The remove topo is done with the assumption that there will only be 1 keysight api server running on a server. The add-topo doesn't have any check to skip creating a new container if one is already present on the server.

Reply:
Made necessary checks to skip `Pull and start Keysight API Server container` task only. `Bind ptf_ip to keysight_api_server` is not skipped so that corresponding `ptf_ip` address of the configuration from testbed is considered each time.

Quick note - For keysight-api-server the container name is constant(i.e. apiserver). The docker_container ansible module "asserts that the container is first present, and then if the container is not running moves it to a running state." Reference [here](https://docs.ansible.com/ansible/latest/collections/community/docker/docker_container_module.html#parameter-state). 

Since the name is constant - multiple keysight-spi-servers are not deployed. There is always only one instant of it as per the existing flow. However required changes have been made accordingly.

Reply:
Hi @neethajohn , does a message that "api-server is already running hence not deploying it again" be good here.

Reply:
Note - 
Since the `Bind ptf_ip to keysight_api_server` task binds the respective ptf_ip from the corresponding configuration from the testbed.csv to the one and only api-server it is necessary to have the `ptf_ip`s same for keysight configurations so that all the testbeds can access that single api-server with the same ip. Using different ips will cause the ips to change accordingly each time add_topo  is done. Api-server is not redeployed. Only the respective ptf_ip is binding is changed.

Reply:
Add topo should not do anything if  the api server docker is already present. We should not change the ptf_ip binding as well. The original ip should be unchanged.

Reply:
One should use the same IP as ptf_ip in testbed.csv for all ixia-deployment in a testbed-server so that it means api-server IP has not changed across testbeds in that server.

For multiple ixia-testbeds if only the first deployed testbeds's ptf_ip is considered for the api-server then ptf_ips of the rest of the testbed entries (if different from the first) will be useless and might create confusion. If same ip used as pft_ips for ixia-testbed entries in the testbed.csv file then all the testbeds can easily use the api server using their respective ptf_ip(which eventually is all same).

Even if the binding is skipped if the api-server is already deployed then also one should keep the ptf_ips same to avoid confusion.

Please let me know if binding is to be skipped if api-server is already deployed.


Following is an example where multiple ixia-testbeds are specified. If we are executing add_topo in a sequence ixia1->ixia2->ixia2 then what ptf_ip entries will be used in actual-
If all the ptf_ips same then there is no problem. If pts_ips are different as in the following example, then say 10.0.0.30 is bound first so it will not be altered. Then what should be considered for 10.0.0.31/10.0.0.32 when ixia2 and ixia3 testbeds are deployed after ixia1.

(Consider the `conf-name` and `ptf_ip` entries only)
```
# conf-name,group-name,topo,ptf_image_name,ptf,ptf_ip,ptf_ipv6,server,vm_base,dut,inv_name,auto_recover,comment
vms-example-ixia-1,vms6-1,t0-64,docker-ptf-ixia,example-ixia-ptf-1,10.0.0.30/32,,server_6,VM0600,example-s6100-dut-1,lab,True,superman
vms-example-ixia-2,vms6-1,t0-64,docker-ptf-ixia,example-ixia-ptf-1,10.0.0.31/32,,server_6,VM0600,example-s6100-dut-2,lab,True,superman
vms-example-ixia-3,vms6-1,t0-64,docker-ptf-ixia,example-ixia-ptf-1,10.0.0.32/32,,server_6,VM0600,example-s6100-dut-3,lab,True,superman
```

Reply:
Ref. [here](https://github.com/Azure/sonic-mgmt/pull/3252#discussion_r607421115).

Changes are implemented. Keysight-api-server will not be altered if already deployed. The original IP would be unchanged.